kafka/kontena.yml

```diff
@@ -0,0 +1,97 @@
+---
+stack: kontena/kafka-cluster
+version: 1.0.0
+description: Kafka cluster based on Harvard Hypermap
+expose: kafka
+variables:
+  link_zookeeper:
+    type: boolean
+    default: true
+    from:
+      env: LINK_ZOOKEEPER
+      prompt: link zookeeper stack?
+  zookeeper_link:
+    type: string
+    only_if: link_zookeeper
+    from:
+      env: LINK_ZOOKEEPER
+      service_link:
+        prompt: Choose zookeeper
+        name: zookeeper
+  zookeeper_uri:
+    type: string
+    default: zookeeper.zookeeper-cluster.${GRID}.kontena.local:9092
+    skip_if: link_zookeeper
+    from:
+      env: ZOOKEEPER_URI
+      prompt: Zookeeper uri
+  num_instances:
+    type: integer
+    min: 1
+    default: 3
+    from:
+      env: NUM_INSTANCES
+      prompt: number of instances of Kafka?
+  expose_kafka_port:
+    type: boolean
+    default: false
+    from:
+      env: EXPOSE_KAFKA_PORT
+  skip_volumes:
+    type: boolean
+    default: false
+    from:
+      env: SKIP_VOLUMES
+  delete_topic_enable:
+    type: boolean
+    default: true
+    from:
+      env: DELETE_TOPIC_ENABLE
+  default_replication_factor:
+    type: integer
+    default: 2
+    from:
+      env: DEFAULT_REPLICATION_FACTOR
+  compression_type:
+    type: string
+    default: lz4
+    from:
+      env: COMPRESSION_TYPE
+
+services:
+  kafka:
+    image: kontena/kafka:latest
+    stateful: true
+    # {% unless skip_volumes %}
+    volumes:
+      - kafka-data:/var/lib/kafka
+    # {% endunless %}
+    instances: ${num_instances}
+    # {% if expose_kafka_port %}
+    ports:
+      - "9092:9092"
+    # {% endif %}
+    # {% if link_zookeeper %}
+    links:
+      - ${zookeeper_link}
+    # {% endif %}
+    environment:
+      # {% assign zookeeper_connect = "" %}
+      # {% if link_zookeeper %}
+      # {%   assign parts = zookeeper_link | split: '/' %}
+      # {%   capture zookeeper_connect %}{{ parts[1] }}.{{ parts[0] }}.{{ GRID }}.kontena.local {% endcapture %}
+      # {% else %}
+      # {%   assign zookeeper_connect = zookeeper_uri %}
+      # {% endif %}
+      KAFKA_ZOOKEEPER_CONNECT: {{ zookeeper_connect }}
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka.${STACK}.${GRID}.kontena.local:9092
+      KAFKA_DELETE_TOPIC_ENABLE: ${delete_topic_enable}
+      KAFKA_DEFAULT_REPLICATION_FACTOR: ${default_replication_factor}
+      KAFKA_COMPRESSION_TYPE: ${compression_type}
+      KAFKA_JMX_PORT: 9999
+# {% unless skip_volumes %}
+volumes:
+  kafka-data:
+    external:
+      name: ${STACK}-kafka-data
```

Comment:
Probably should be just `${STACK}-data` (no need to repeat kafka)?

Reply:
Yeah good call... was a leftover from both services being in same stack originally.